runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -519,20 +538,21 @@ MM_CopyForwardScheme::postProcessRegions(MM_EnvironmentVLHGC *env)
 			Assert_MM_false(region->_reclaimData._shouldReclaim);
 
 			/* we do not count tails, only regions that we acquired as free */
-			if (region->getLowAddress() == region->_copyForwardData._survivorBase) {
+			if (region->isFreshSurvivorRegion()) {
 				survivorSetRegionCount += 1;
 			}
 			/* store back the remaining memory in the pool as free memory */
-			UDATA remainingBytes = pool->getAllocatableBytes();
+//			UDATA remainingBytes = pool->getAllocatableBytes();
 
 			region->_sweepData._alreadySwept = true;
-			pool->setFreeMemorySize(pool->getActualFreeMemorySize() + remainingBytes);
-			UDATA holeCount = (0 == remainingBytes) ? 0 : 1;
-			pool->setFreeEntryCount(holeCount);
-			pool->setLargestFreeEntry(remainingBytes);
-			Assert_MM_true(pool->getActualFreeMemorySize() >= pool->getAllocatableBytes());
-			Assert_MM_true(pool->getActualFreeMemorySize() <= region->getSize());
-			if (pool->getActualFreeMemorySize() == region->getSize()) {
+//			pool->setFreeMemorySize(pool->getActualFreeMemorySize() + remainingBytes);
+//			UDATA holeCount = (0 == remainingBytes) ? 0 : 1;
+//			pool->setFreeEntryCount(holeCount);
+//			pool->setLargestFreeEntry(remainingBytes);
+//			Assert_MM_true(pool->getActualFreeMemorySize() >= pool->getAllocatableBytes());
+//			Assert_MM_true(pool->getActualFreeMemorySize() <= region->getSize());
+			if (pool->getFreeMemoryAndDarkMatterBytes() == region->getSize()) {
```

Comment:
comment below should be updated (not use BUMP_ALLOCATED, but AOL or whatever is appropriate)